src/main/kotlin/vm/VM.kt

```diff
@@ -77,15 +88,32 @@ class VM(
             if (dropReturnValue) dropReturnValue = false
             else push(it)
         }
-        try {
-            return executeCode()
-        } catch (e: Exception) {
-            throw (e as? VMException ?: VMException(E_SYS,
-                "${e.message} (mem $pc)\n${e.stackTraceToString()}"
-            )).withLocation(lineNum, charNum)
+        while (true) {
```

Comment:
Much like in Task, we need to repeat VM execution forever, in case the catch block catches locally and we need to continue.